- restore freshon search plugin from 8636793fe6532fee086ec6664eefb844…

Jesus-21 opened the pull request:
### Motivation for changes:
Freshon.tv search plugin is documented but was not available.
I guess it has not been ported when the python3 compatibility has been done.

### Detailed changes:

- I restored the plugin from commit 8636793fe6532fee086ec6664eefb844
- moved the file from `flexget/plugins/search_freshon.py` to `flexget/plugins/sites/freshon.py` 
- As advised in https://flexget.com/Developers#python23-compatibility I added from the original file
```python
from builtins import *
```
- Only change from original plugin is that I removed `verify=False` the post request

### Config usage if relevant (new plugin or updated schema):
```
tv_search_freshon:
    discover:
      what:
         - trakt_list:
             username: xxxxxxx
             password: xxxxxxx
             list: somelist
             type: shows
          from:
            - freshon:
                username: xxxxxx
                password: xxxxxx
                passkey: xxxxx
```
Note: `passkey` is not present in documentation but is/was mandatory (cf. https://flexget.com/Searches/freshon )

cvium commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                        size = re.search('(\d+(?:[.,]\d+)*)\s?([KMG]B)', size)
+
+                        if size:
+                            if size.group(2) == 'GB':
```
https://github.com/Flexget/Flexget/blob/develop/flexget/utils/tools.py#L437

cvium commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                        if (int(entry['torrent_seeds']) > 0):
+                            entries.add(entry)
+                        else:
+                            log.debug('0 SEEDS, not adding entry')
```
I'm a little puzzled by this and I assume it was always there. That should not be up to the search plugin decide. We already have https://flexget.com/Plugins/torrent_alive if people don't want dead torrents.

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+            'freeleech': {
+                'type': 'string',
+                'default': 'all',
+                'enum': ['free', 'half', 'all']
```
Best to use this as a dict for reusability:
```python
LEECHSTATUS = {
    'all': 0,
    'free': 3,
    'half': 4
}
```
then in the schema use:
```python
'enum': list(LEECHSTATUS.keys())
```
and later you can access this dict too

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                      'login': 'Do it!'}
+            urlstr = (base_url, login_fragment)
+            lsrc = task.requests.post('/'.join(urlstr), data=params)
+            if str(config['username']) in lsrc.text:
```
no need for `str`

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                log.error('Login to FreshonTV was NOT successful')
+                raise plugin.PluginError("Login to FreshonTV was NOT successful")
+
+        if config['freeleech'] == 'all':
```
See above about reusing dicts

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+        for search_string in entry.get('search_strings', [entry['title']]):
+            query = normalize_unicode(search_string.replace(' ', '+'))
+            query_url_fragment = urllib.quote_plus(query.encode('utf-8'))
+            str_url = (base_url, search_fragment)
```
no need for `str`, separate urls and params

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+            str_url = (base_url, search_fragment)
+            url = '/'.join(str_url) % (query_url_fragment, config['category'], str(freeleech))
+            log.debug('Search url: %s' % url)
+            page = task.requests.get(url).content
```
best to wrap this in a try, except with `RequestsException`

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                while (nextpage < max_page_number):
+                    if (nextpage > 0):
+                        newurl = url + '&page=' + str(nextpage)
+                        log.debug('-----> NEXT PAGE : %s' % newurl)
```
pass string interpolation to logger, replace `%` with `,`:
` log.debug('-----> NEXT PAGE : %s', newurl)`

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+                    for res in soup.findAll('tr', {'class': re.compile('torrent_[0-9]*')}):
+                        entry = Entry()
+                        # skip if nuked
+                        if res.find('img', alt='Nuked') != None:
```
maybe add log for this scenario?

liiight commented on flexget/plugins/sites/freshon.py:
```diff
@@ -0,0 +1,188 @@
+
+@event('plugin.register')
+def register_plugin():
+    plugin.register(SearchFreshon, 'freshon', groups=['search'], api_ver=2)
```
change `groups` to `interfaces`

Jesus-21 commented:
It needed a slight review indeed, I cleaned it and made sure it was PEP8 compliant in the process.
I also added `page_limit` to the config, hardcoding a limit to 10 pages seemed a bad idea to me.

cvium commented on flexget/plugins/sites/freshon.py:
```diff
@@ -9,10 +9,21 @@
+BASE_URL = 'https://freshon.tv'
+SEARCH_FRAGMENT = 'browse.php?search=%s&tab=%s&incldead=%s'
+DL_FRAGMENT = 'download.php?type=rss&id=%s&passkey=%s'
+LOGIN_FRAGMENT = 'login.php?action=makelogin'
```
Can't all these "fragments" be replaced with dicts instead? Like you're doing in `login`.

cvium commented on flexget/plugins/sites/freshon.py:
```diff
@@ -176,8 +181,13 @@ def parse_entry(self, res):
-        entry['url'] = '/'.join(url) % (id, self.config['passkey'])
+        params = {
+            'type': 'rss',
+            'id': id,
```
`id` is a python keyword. Please call it something else.

liiight commented:
Looks good to me
